### DISCUSSION TITLE
Providing Credentials to Gradle in Renovate hosted app on github.com

### DISCUSSION THREAD

**Discussion by yetanotherion** [renovatebot/renovate#20473] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Same as https://github.com/renovatebot/renovate/discussions/19363 but for Renovate hosted app on github.com.
What's the best way to:
- setup env variables, containing secret values, 
- and pass them to the docker executed by the renovatebot?

According to the doc, **customEnvVariables** are only available for self hosted apps.
And before being able to use **exposeAllEnv**, I still did not find a way to define the env variables.

### What you tried so far.

In our setup, gradle gets those creds as
```
maven {
            url = uri("https://maven.pkg.github.com/<private_url>")
            credentials {
                username = System.getenv("GITHUB_PACKAGE_USERNAME")
                password = System.getenv("GITHUB_PACKAGE_TOKEN")
            }
        }
```

We have **hostRules** setup in _renovate.json_, 
```
"hostRules": [
    {
      "matchHost": "https://maven.pkg.github.com/<private_url>",
      "username": "<username>",
      "encrypted": {
        "token": <token>
      }       
]
```
which work great for scanning repositories on the onboarding PR.

In the post onboarding PRs that update gradle dependencies, renovatebot fails

as **GITHUB_PACKAGE_USERNAME** and **GITHUB_PACKAGE_TOKEN** are not set (see logs below).

### Relevant debug logs

<details><summary>Logs</summary>

```
Command failed: docker run --rm --name=renovate_sidecar --label=renovate_child -v "/mnt/renovate/gh/<project>":"/mnt/renovate/gh/<project>" -v "/tmp/renovate-cache":"/tmp/renovate-cache" -v "/tmp/containerbase":"/tmp/containerbase" -e GRADLE_OPTS -e BUILDPACK_CACHE_DIR -e CONTAINERBASE_CACHE_DIR -w "/mnt/renovate/gh/<project>" docker.io/renovate/sidecar bash -l -c "install-tool java 16.0.2+7 && ./gradlew --console=plain -q properties"
openjdk version "16.0.2" 2021-07-20
OpenJDK Runtime Environment Temurin-16.0.2+7 (build 16.0.2+7)
OpenJDK 64-Bit Server VM Temurin-16.0.2+7 (build 16.0.2+7, mixed mode, sharing)

FAILURE: Build failed with an exception.

* Where:
Build file '/mnt/renovate/gh/.../build.gradle.kts' line: 10

* What went wrong:
Plugin [id: '...', version: '...'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Plugin Repositories (could not resolve plugin artifact '...')
  Searched in the following repositories:
    Gradle Central Plugin Repository
    maven(https://maven.pkg.github.com/<private_url>)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2m 57s
```

</details>

**Comment by viceice** [comment, marked as answer]
not yet supported

**Reply by yetanotherion** [reply]
Thanks @viceice!

**Reply by gmariottiAtWolt** [reply]
@viceice is there a plan to support it or will it remain an option for self-hosted renovate?

**Reply by viceice** [reply]
not yet, somebody needs to write a summary how this can be solved and then open a feature request with a reproduction.

**Reply by gmariottiAtWolt** [reply]
Is this something that I could do? I think I can create the feature request with an example repository showing the problem, but I'm not sure how I can figure out why it doesn't work in Mend Renovate

**Reply by viceice** [reply]
- https://github.com/renovatebot/renovate/issues/11061
